src/test/java/PlayListTests.java

```diff
@@ -0,0 +1,89 @@
+
+import org.testng.Assert;
+import org.testng.annotations.Test;
+import pages.LoginPage;
+import pages.PlayListPage;
+
+public class PlayListTests extends BaseTest{
+
+    @Test
+    public void addSongToPlaylist() throws InterruptedException {
+
+        // create a playlist
+        createPlayListBtn(playListName);
+        // search for a song
+        Thread.sleep(3000);
+        searchSong(songName);
+        // click view all button
+        viewAllSongs();
+        // select first song from result
+
+        selectFirstSong();
+        // click Add to Button
+        addSongToBtn();
+        // choose list to add song to
+        Thread.sleep(3000);
+        choosePlayList();
+        Thread.sleep(5000);
+        deletePlayListBtn();
+    }
```

Comment:
Please, delete `throws InterruptedException`

**Don't use Thread.sleep! It's not a good way to wait for things to happen, especially in tests. It can make your program act weird. Try other ways, like explicit waiting. It's more reliable for tests.**

Here's why you should avoid it:

**Unreliable Timing:** Using a fixed sleep duration disregards the actual time needed for an action to complete. System speed, network connectivity, or server load can all affect the execution time, resulting in false negatives or hiding real issues. It's crucial to have tests that accurately reflect the system's behavior.

**Reduced Efficiency:** Thread.sleep() adds unnecessary delays to test execution. Even if an action completes before the specified sleep time, the thread remains idle, prolonging the overall test duration. For a large number of waits, this can significantly impact your testing efficiency.

**Lack of Synchronization:** Thread.sleep() doesn't consider the actual state of your application or UI elements under test. You may encounter situations where the expected element is not yet available, even after the specified sleep time. This can lead to false results or longer waits than necessary.

By embracing explicit waits, you can enhance the reliability, efficiency, and synchronization of your tests, leading to more robust automation. Say goodbye to hidden bugs and embrace a better testing approach!

**PS: In real work, Thread.sleep is hardly ever used. I recommend refactoring all your methods without using Thread.sleep.**

**Use these methods in the BasePage, and then in child classes**

**Waite**

```
  public WebElement waitForPresence(By by) {
    return wait.until(ExpectedConditions.presenceOfElementLocated(by));
  }

  public List<WebElement> waitForPresenceOfAllElements(By by) {
    return wait.until(ExpectedConditions.presenceOfAllElementsLocatedBy(by));
  }

  public WebElement waitForVisibility(By by) {
    return wait.until(ExpectedConditions.visibilityOfElementLocated(by));
  }

  public List<WebElement> waitUntilAllElementsLocatedByAreVisible(By by) {
    return wait.until(ExpectedConditions.visibilityOfAllElementsLocatedBy(by));
  }

  public boolean waitUntilRequiredElementsAreInvisible(List<WebElement> elements) {
    return wait.until(ExpectedConditions.invisibilityOfAllElements(elements));
  }

  public WebElement waitForClickable(By by) {
    return wait.until(ExpectedConditions.elementToBeClickable(by));
  }
```

**Click**

```
  public void click(By by) {
       waitForClickable(by).click();
  }

  public void waitForClickableAndClick(By by) {
    WebElement element = waitForClickable(by);
    element.click();
  }

  public void tryClick(By by) {
    waitForVisibility(by).click();
  }

  public void tryClickNoWait(By by) {
     driver.findElement(by).click();
  }

  public void doubleClick(By by) {
    WebElement element = waitForClickable(by);
    actions.doubleClick(element).perform();
  }


  public void mouseOverAndClick(By by) {
    WebElement element = driver.findElement(by);
    actions.moveToElement(element).click().perform();
 
  }

  public void mouseOverAndClickAndHold(By by) {
    WebElement element = driver.findElement(by);
    actions.moveToElement(element).clickAndHold().perform();
 
  }
```


PS: _Suppose you have written Thread.sleep(1000) in all places in your code. But the next day the load on the testing system has increased, and now you need to change all values to 1500. How many gray hairs will you get while you do that? LOL -)_